junit-jupiter-api/src/main/java/org/junit/jupiter/api/MethodOrderer.java

```diff
@@ -172,6 +175,76 @@ private static int getOrder(MethodDescriptor descriptor) {
 		}
 	}
 
+	/**
+	 * {@code MethodOrderer} that sorts methods based on the {@link Depend @Depend}
+	 * annotation.
+	 */
+	class DependAnnotation implements MethodOrderer {
+		private static final Logger logger = LoggerFactory.getLogger(Random.class);
+
+		private static Map<String, Integer> dependencySize;
+		private static Map<String, String[]> digraph;
+
+		@Override
+		public void orderMethods(MethodOrdererContext context) {
+			digraph = context.getMethodDescriptors().stream().filter(
+				descriptor -> descriptor.findAnnotation(Depend.class).isPresent()).collect(
+					Collectors.toMap(descriptor -> descriptor.getMethod().getName(),
+						descriptor -> descriptor.findAnnotation(Depend.class).map(Depend::methods).get()));
+
+			dependencySize = new HashMap<>();
+
+			try {
+				for (String name : digraph.keySet()) {
+					if (!dependencySize.containsKey(name)) {
+						int sz = DFS(name);
+						if (sz == -1) {
+							throw new IllegalArgumentException("Cycle Detected!!!");
+						}
+					}
+				}
+			}
+			catch (IllegalArgumentException exception) {
+				logger.error(exception,
```

Comment:
I would argue that the exception should not be swallowed here.

If the user has a configuration error in the test class, that should fail the test class.

Please rework this to throw a `JUnitException` instead.

Reply:
I take that back.

A `MethodOrderer` is applied during the "discovery phase", thus any such exception will be swallowed anyway and only logged at `DEBUG` level by the Jupiter test engine.

Thus, it's best to log it as an error here as you have done.